src/main/kotlin/org/rust/ide/inspections/RustDropRefInspection.kt

```diff
@@ -0,0 +1,45 @@
+package org.rust.ide.inspections
+
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.psi.PsiElementVisitor
+import org.rust.ide.inspections.fixes.RemoveRefFix
+import org.rust.lang.core.psi.RustCallExprElement
+import org.rust.lang.core.psi.RustElementVisitor
+import org.rust.lang.core.psi.RustPathExprElement
+import org.rust.lang.core.psi.RustUnaryExprElement
+import org.rust.lang.core.psi.impl.RustFnItemElementImpl
+
+/**
+ * Checks for calls to std::mem::drop with a reference instead of an owned value. Analogue of Clippy's drop_ref.
+ * Quick fix: Use the owned value as the argument.
+ */
+class RustDropRefInspection : RustLocalInspectionTool() {
+    override fun getDisplayName(): String = "Drop reference"
+
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor {
+        return object : RustElementVisitor() {
+            override fun visitCallExpr(expr: RustCallExprElement) {
+                inspectExpr(expr, holder)
+            }
+        }
+    }
+
+    fun inspectExpr(expr: RustCallExprElement, holder: ProblemsHolder) {
+        val pathExpr = expr.expr
+        if (pathExpr is RustPathExprElement) {
+            val resEl = pathExpr.path.reference.resolve()
+            if (resEl is RustFnItemElementImpl && resEl.canonicalCratePath.toString() == "::mem::drop") {
+                val args = expr.argList.exprList
+                if (args.size == 1) {
+                    val arg = args[0]
+                    if (arg is RustUnaryExprElement && arg.expr is RustPathExprElement) {
```

Comment:
Hey, this can be more precise and catch cases like

```
let x = &foo();
drop(x)
```

if you look at the type of the expression: `if (arg.resolvedTye is RustReferenceType)`. The entry to the type machinery is [here](https://github.com/intellij-rust/intellij-rust/blob/3dd1a3f922f8f59ccfe6b2705781cfb82f2784d4/src/main/kotlin/org/rust/lang/core/types/util/RustPsiExtensions.kt#L12). It is very WIP at the moment, but it should be strictly better than checking for literal `&` (if it is not, it is a bug :) ).